### PR TITLE
Feature/excel header identity

### PR DESCRIPTION
**Pull request by manas-droid**
1. Added a new Input Control to accept Table Header Row Number
2. Added precise error messages

Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [x] Existing unit test cases are passed
- [x] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [x] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/ActionsLib/ActExcel.cs [hunk 1]
```diff
@@ -357,7 +390,14 @@ public void ReadData()
             }
             catch (Exception ex)
             {
-                Error += eUserMsgKey.ExcelInvalidFieldData + ", " + ex.Message;
+                if(ex.Message!=null && ex.Message.StartsWith("Cannot find column"))
+                {
+                    Error = ex.Message + " " + $"at Row Number {HeaderRowNum}";
```
[the remark this answers is not in the document]
Comment (manas-droid) [reply Ginger/GingerCoreNET/ActionsLib/ActExcel.cs]: Implemented the requested changes

File: Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs [hunk 2]
```diff
@@ -39,60 +39,36 @@ public class ExcelNPOIOperations : IExcelOperations
...
-            {
-                Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "Can't convert sheet to data, " + dupEx.Message);
-                throw;
-            }
             catch (Exception ex)
             {
-                Reporter.ToLog(eLogLevel.WARN, "Can't convert sheet to data, " + ex.Message);
-                return null;
+                string exceptionMessage = $"Can't convert sheet to data, {ex.Message}";
+                Reporter.ToLog(eLogLevel.WARN, exceptionMessage);
```
[the remark this answers is not in the document]
Comment (manas-droid) [reply Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs]: The requested change was implemented in the latest code

File: Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs [hunk 3]
```diff
@@ -161,15 +153,14 @@ private bool GetExcelSheet(string fileName, string sheetName)
                 if (mWorkbook == null)
                 {
                     Reporter.ToLog(eLogLevel.WARN, "File name not Exists.");
-                    return false;
+                    throw new ArgumentException("File DOES NOT Exist or is currently being used by some other application, Please verify if the File Path is valid");
```
[the remark this answers is not in the document]
Comment (manas-droid) [reply Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs]: The requested change was implemented in the latest code

File: Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs [hunk 4]
```diff
@@ -221,82 +212,91 @@ public bool UpdateExcelData(string fileName, string sheetName, string filter, Li
...
-            }
-            var i = cellFrom.Row;
-            var currentRow = mSheet.GetRow(i);
-            int dtColCount = 0;
-            while (i <= cellTo.Row)
-            {
-                var dr = dtExcelTable.NewRow();
-                dtColCount = 0;
-                for (var j = cellFrom.Col; j <= cellTo.Col; j++)
+                Regex regex = new Regex(@"(^[A-Z]+\d+$)|(^[A-Z]+\d+:[A-Z]+\d+$)");
```
[the remark this answers is not in the document]
Comment (manas-droid) [reply Ginger/GingerCoreNET/Run/Excels/ExcelNPOIOperations.cs]: The requested change was implemented in the latest code

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**